lib/src/tag_detection.dart

```diff
@@ -640,140 +742,132 @@ class Tagger {
     final isBinaryOnly = nonSrcDartFiles.isEmpty && allBinFiles.isNotEmpty;
 
     return Tagger._(
-        packageDir, session, pubspecCache, isBinaryOnly, topLibraries);
+        pubspec.name, session, pubspecCache, isBinaryOnly, topLibraries);
   }
 
-  Set<String> declaredSdks(String packageDir) {
-    return {
-      ..._pubspecCache.pubspecOfPackage(packageDir).dependentSdks,
-      'dart'
-    };
-  }
-
-  /// Returns `true` iff the package at [packageDir] supports [sdk].
-  bool _supportsSdk(Sdk sdk) {
+  void sdkTags(List<String> tags, List<Suggestion> suggestions) {
     if (_isBinaryOnly) {
-      return sdk.name == 'dart';
-    }
-    if (_topLibraries.isEmpty) {
-      return false;
+      tags.add('sdk:dart');
+    } else if (_topLibraries.isEmpty) {
+      suggestions.add(
+        Suggestion.hint(
+            SuggestionCode.noToplevelLibraries,
+            'No top-level libraries found',
+            'Cannot assign sadk tags, because no .dart files where found in lib/'),
+      );
+    } else {
+      for (final sdk in Sdk.knownSdks) {
+        // Will find a path in the package graph where a package declares an sdk
+        // not supported by [sdk].
+        final violationResult =
+            SdkViolationFinder(_packageGraph, sdk, _pubspecCache, _session)
+                .findSdkViolation(packageName, _topLibraries);
+        if (violationResult != null) {
+          suggestions.add(violationResult);
+        } else {
+          tags.add(sdk.tag);
+        }
+      }
     }
-    // Will find a path in the package graph where a package declares an sdk
-    // not supported by [sdk].
-    final declaredSdkViolationFinder =
-        _PathFinder(_packageGraph, (String packageDir) {
-      return !declaredSdks(packageDir).every(sdk.allowedSdks.contains);
-    });
+  }
 
-    final pathResult = declaredSdkViolationFinder.findPath(packageDir);
-    if (pathResult.hasPath) {
-      log.info(
-          '$packageDir does not support ${sdk.name} because of the package: ${pathResult.path}');
-      return false;
+  /// Adds tags for the Flutter platforms that this package supports to [tags].
+  ///
+  /// Adds [Suggestion]s to [suggestions] for platforms not supported.
+  void flutterPlatformTags(List<String> tags, List<Suggestion> suggestions) {
+    if (_isBinaryOnly) {
```

Comment:
please comment why it is empty and why that case is ignored

Reply:
Hmm - I think I was unsure what to write, postponed it and forgot ...